src/services/TemplatingService.ts

```diff
@@ -0,0 +1,150 @@
+          aps: {
+            title: payload.title,
+            body: payload.body,
+          },
+        },
+      },
+    });
+  }
+
+  private static insertUserPayload(template: any, user: UserRessource): any {
+    // TODO: Access deviceId of User
+    return Object.assign(template, {
+      to: `${user.name} <${user.mail}>`,
+      token: 'user.deviceToken',
+    });
+  }
+
+  // endregion
+
+  // region public members
+  // endregion
+
+  // region private members
+
+  private readonly localizedTemplates: Template[] = [];
+
+  // endregion
+
+  // region constructor
+
+  public constructor(platformId: string, templateId: string, payload: any) {
+    this.generateTemplates(platformId, templateId, payload);
+  }
+
+  // endregion
+
+  // region public methods
+
+  public createMailMessage(user: UserRessource): Mail {
+    const template = this.getTemplateForUser(user, MAIL_MESSAGE);
+    const mail = {
+      from: template.from,
+      to: user.mail,
+      subject: template.subject,
+      text: template.text,
+      html: template.html,
+    };
+    return mail;
+  }
+
+  public createPushMessage(user: UserRessource): firebaseMessaging.Message {
+    const template = this.getTemplateForUser(user, PUSH_MESSAGE);
+    const push = {
+      token: template.token,
+      data: template.data,
+      notification: template.notification,
+      android: template.android,
+      webpush: template.webpush,
+      apns: template.apns,
+    };
+    return push;
+  }
+
+  // endregion
+
+  // region private methods
+
+  private generateTemplates(platformId: string, templateId: string, payload: TemplatePayload[]) {
+    for (const type of [MAIL_MESSAGE, PUSH_MESSAGE]) {
+      // Step 1: Load base template
+      let baseTemplate = Utils.getTemplate(platformId, templateId, type);
+
+      // Step 2: Insert general payload into base template
+      // TODO: Pass general message payload (with e.g. unique language identifier)
+      baseTemplate = TemplatingService.insertMessagePayload(baseTemplate, {}, platformId);
+
+      // Step 3: Insert language specific payload into base template for every language
+      for (const languagePayload of payload) {
+        const template = TemplatingService.insertLanguagePayload(baseTemplate, languagePayload.payload);
+        const languageId = languagePayload.language;
+        const localizedTemplate = {
+          languageId,
+          type,
+          template,
+        };
+        this.localizedTemplates.push(localizedTemplate);
+      }
+    }
+  }
+
+  private getTemplate(languageId: string, type: string): any {
+    const currentTemplate: Template | undefined = this.localizedTemplates.find(
+      (template: Template) => {
+        return (template.languageId === languageId) && (template.type === type);
+      },
+    );
+
+    if (currentTemplate) {
+      return currentTemplate.template;
+    } else {
+      return Promise.reject(new Error('Invalid languageId or type. Template not found.'));
```

Comment:
There is no need to return a promise here, because all function calles inside `getTemplate` are synchronous.

Reply:
Done